### ISSUE TITLE
Ask for smoothness on segregated Cycle & Foot paths

### ISSUE THREAD

**Issue by rhhsm** [streetcomplete/StreetComplete#5451]
For segregated Cycle & Foot paths (highway=cycleway/footway/path + bicycle=designated + foot=designated + segregated=yes), SC currently asks for `surface` of both the cycle path and the foot path. However, these kind of paths are not yet included in the `smoothness` quest. I see no reason why they shouldn't be included, i.e. for highway=cycleway/footway/path that have `cycleway:surface=*` and/or `footway:surface=*`, `smoothness` should be quested so that `cycleway:smoothness` and `footway:smoothness` are set. The same UI that is used for the surface quest for these paths can be used.

In addition, the Dutch tagging tradition of seeing these kind of paths as a cycle path with a sidewalk for pedestrians should also be taken into account, i.e. ways like this one https://www.openstreetmap.org/way/6547886 should also be quested for the smoothness of both cycle path and sidewalk.

There may be other properties like `width` (that I'm not tagging) that are not quested for these kinds of highways (that's why I thought my idea was related to #5445)

**Comment by westnordost** [comment]
Minus the note about sidewalks, this could be an enhancement of the smoothness quest.

**Comment by rhhsm** [comment]
> Minus the note about sidewalks

Why? Maybe because you're afraid the next thing I'm going to suggest is to ask for smoothness for all sidewalks? I've been thinking about that, but it would be an awful lot of work to survey... ;)

**Comment by Helium314** [comment]
Asking for `sidewalk:surface` _is_ already asking smoothness for all sidewalks, be it on a road or on a cycleway.
Though this case is unclear to me, is it really a segragated cycleway with sidewalk? I.e. two lanes for foot where the foot smoothness should be asked twice?

**Comment by rhhsm** [comment]
> Asking for `sidewalk:surface` _is_ already asking smoothness for all sidewalks

If `smoothness `would be predictable from `surface` then there would be no need for the `smoothness `tag. In reality, any surface can vary widely in smoothness to a degree that a highway can be impassable for certain vehicles, and that's why the smoothness tag was created.

> Though this case is unclear to me, is it really a segragated cycleway with sidewalk? I.e. two lanes for foot where the foot smoothness should be asked twice?

The way I quoted above is one lane for bicycles and one for foot (see https://maps.app.goo.gl/6d5EUCsrRAfk1kP5A left=foot, right=cycle). However the Dutch OSM community decided to tag such a path as a cycleway with sidewalk instead of a segregated way for cycles and pedestrians. They are the same thing in reality so they should be treated in the same way by SC.

**Comment by Helium314** [comment]
> Dutch OSM community decided to tag such a path as a cycleway with sidewalk **instead** of a segregated way for cycles and pedestrians

So the tagging of the way you linked is actually wrongly tagged?

**Comment by rhhsm** [comment]
> > Dutch OSM community decided to tag such a path as a cycleway with sidewalk **instead** of a segregated way for cycles and pedestrians
> 
> So the tagging of the way you linked is actually wrongly tagged?

It's tagged according to how the Dutch community decided it should be tagged. If you think that decision is wrong, please take it up with the Dutch community. I don't think this is the place to discuss it.

**Comment by Helium314** [comment]
If such duplicate surface tagging is seen as correct, SC smoothness quests for both `sidewalk:surface` and `footway:surface` will certainly be a waste of user's time.

**Comment by rhhsm** [comment]
> If such duplicate surface tagging is seen as correct, SC smoothness quests for both `sidewalk:surface` and `footway:surface` will certainly be a waste of user's time.

It's not duplicate, the footway's surface is tagged either as `sidewalk:surface` (in NL) or as `footway:surface` (rest of the world), not both

**Comment by Helium314** [comment]
You linked to a way that has `sidewalk:surface` __AND__ `footway:surface` and confirmed that tagging is correct and not duplicate, and then claim it's __EITHER__ `sidewalk:surface` __OR__ `footway:surface`.

I really don't get it, sorry.

**Comment by rhhsm** [comment]
> I'd say, don't waste your time.

I'll give it a try, so you don't have to waste your time adapting SC to the Dutch tagging.

Please mark most of the above discussion as off topic :)

**Comment by westnordost** [comment]
By the way, I just noticed this thread: https://community.openstreetmap.org/t/fietspad-als-straat/108345

**Comment by westnordost** [comment]
Now you are wasting my time too, making it look like StreetComplete does not handle the Dutch sidewalks-on-cycleways tagging scheme correctly and me having to clarify it. :-/

**Comment by mcliquid** [comment]
Here is my idea:

The two-column “Foot Path / Cycle Path” surface UI is actually part of the Surface overlay, while the corresponding quests are implemented separately as AddFootwayPartSurface and AddCyclewayPartSurface.

For smoothness, reusing the overlay UI in a quest would require a new quest-specific pair selector and additional changes because the smoothness image set depends on the respective footway:surface / cycleway:surface.

So the smallest implementation would be two sibling quests analogous to the existing part-surface quests:

AddFootwayPartSmoothness → footway:smoothness=*
AddCyclewayPartSmoothness → cycleway:smoothness=*

Both would reuse the existing smoothness form with minor parameterization.

Questions:
- Would this two-quest approach be preferred here for?
- Should "is actually steps" and "surface is different" be offered at all on segregated paths? Both are destructive with prefixed tags; I'd omit them.
- segregated=yes with only plain surface, and footway:surface without segregated=yes, currently get no smoothness quest. Intentional, or widen?
- Should a stale unprefixed smoothness be removed once both part values are set, by analogy with updateCommonSurfaceFromFootAndCyclewaySurface? There is no "common smoothness" concept today.

Out of Scope:
- Dutch sidewalk:*

**Comment by westnordost** [comment]
Yes, if it is implemented as a quest, then that is the way to go.

However, `AddSmoothness` has long been on my "should actually be an overlay and the quest be deleted" list. Even if in the end, when a Smoothness overlay is finally implemented, the quest is not removed, it is the question whether it is worth expending time and effort in an approach that is clearly inferior to just having this as an overlay.

> segregated=yes with only plain surface, and footway:surface without segregated=yes, currently get no smoothness quest. Intentional, or widen?

Don't quite follow. Example?

> Should a stale unprefixed smoothness be removed once both part values are set, by analogy with updateCommonSurfaceFromFootAndCyclewaySurface? There is no "common smoothness" concept today.

Makes sense to me.

**Comment by mcliquid** [comment]
Thanks, that makes sense. I don't like the idea of a smoothness-overlay-only but in that case I agree that spending effort on two new quest types may not be the best direction if a Smoothness overlay is the intended long-term solution.

For the example I meant:

highway=path + bicycle=designated + foot=designated + segregated=yes + surface=asphalt, but without cycleway:surface / footway:surface. AddPathSmoothness excludes it because it is segregated, while part-specific smoothness quests based on the corresponding part surface would not match either.
Similarly, a path may already have cycleway:surface=asphalt and footway:surface=paving_stones but no explicit segregated=yes. The Surface overlay already interprets the part-specific surface tags as separated surfaces.

Given your comment, I'd rather look into #5486 and investigate implementing a Smoothness overlay instead of adding two new smoothness quests. But as I said above, I'd be heartbroken if the quest disappeared. Maybe I can implement both at the same time in SCEE.

For stale unprefixed smoothness, agreed — if both part-specific smoothness values are set, removing the common value sounds consistent.
